File: examples/node/image-generation.mjs
import { BeatAPIClient } from "./lib/beatapi.mjs";

const client = new BeatAPIClient();
const task = await client.createImageTask({
  model: "nano-banana",
  prompt: "Editorial product photograph on a warm stone pedestal.",
  aspect_ratio: "1:1",
  output_format: "png",
});

console.log(`Created ${task.id}`);
console.log(JSON.stringify(await client.waitForTask(task.id), null, 2));

File: examples/python/image_generation.py
import json

from beatapi import BeatAPIClient


client = BeatAPIClient()
task = client.create_image_task(
    {
        "model": "nano-banana",
        "prompt": "Editorial product photograph on a warm stone pedestal.",
        "aspect_ratio": "1:1",
        "output_format": "png",
    }
)
print(f"Created {task['id']}")
print(json.dumps(client.wait_for_task(task["id"]), indent=2))
